src/HW/client/src/components/AppContainer/AppContainer.jsx

```diff
@@ -1,85 +1,38 @@
-import ListControls from "../ListControls/ListControls";
-import ListFooter from "../ListFooter/ListFooter";
-import {
-  fetchItems,
-  createItem,
-  deleteItem,
-  updateStatus,
-  editTaskName,
-  deleteAllItems,
-} from "../../Services/ItemClient";
-
-function AppContainer() {
-  const [items, setItems] = useState([]);
-  const [numOfTasks, setNumOfTasks] = useState(0);
-
-  const renderNewItems = async (item) => {
-    try {
-      const newItems = await createItem(item);
-      newItems.forEach((item) => {
-        items.push(item);
-      });
-      setItems([...items]);
-      setNumOfTasks(items.length);
-    } catch (err) {
-      throw new Error(err);
-    }
-  };
-
-  const deleteItemFromTodoList = async (itemId) => {
-    try {
-      await deleteItem(itemId);
-      const idx = items.findIndex((item) => item.itemId === itemId);
-      items.splice(idx, 1);
-      setItems([...items]);
-      setNumOfTasks(items.length);
-    } catch (err) {
-      throw new Error(err);
-    }
-  };
-  const clearAllFromDb = async () => {
-    try {
-      await deleteAllItems();
-      setItems([]);
-      setNumOfTasks(0);
-    } catch (err) {
-      throw new Error(err);
-    }
-  };
+import ListConnector from "../List/ListConnector";
+import ListControlsConnector from "../ListControls/ListControlsConnector";
+import ListFooterConnector from "../ListFooter/ListFooterConnector";
+import PropTypes from "prop-types";
+import SearchBoxConnector from "../SearchBox/SearchBoxConnector.js";
+import SelectBoxConnector from "../SelectBox/SelectBoxConnector";
+import DoneTasksConnector from "../DoneTasks/DoneTasksConnector";
 
+const AppContainer = ({ numOfItems, getItemsAction, tasksStatusState }) => {
   useEffect(() => {
-    fetchItems().then((fetchedItems) => {
-      setItems(fetchedItems);
-      setNumOfTasks(fetchedItems.length);
-    });
-  }, []);
+    getItemsAction();
+  }, [getItemsAction]);
 
   return (
     <section className="main-section">
       <div className="todoApp" id="todoApp">
         <h1 className="">My Todo List</h1>
 
         <div>
-          <ListControls renderNewItems={renderNewItems} />
-          <List
-            items={items}
-            deleteItemFromDb={deleteItemFromTodoList}
-            updateStatusDb={updateStatus}
-            editTaskNameDb={editTaskName}
-          />
-          {numOfTasks > 0 && (
-            <ListFooter
-              numOfTasks={numOfTasks}
-              clearAllFromDb={clearAllFromDb}
-            ></ListFooter>
-          )}
+          <ListControlsConnector />
+          {numOfItems > 0 && <SearchBoxConnector />}
+          {tasksStatusState ? <ListConnector /> : <DoneTasksConnector />}
+          {numOfItems > 0 && <SelectBoxConnector />}
+          {numOfItems > 0 && <ListFooterConnector />}
         </div>
       </div>
     </section>
   );
-}
+};
+AppContainer.propTypes = {
+  numOfItems: PropTypes.number,
+  getItemsAction: PropTypes.func,
+  tasksStatusState: PropTypes.bool,
```

Comment:
the name of the variable is not clear. state for something that represents a boolean 
also its better practice to change to enum in case you have more states in the future